typescript/agentkit/src/wallet-providers/privyWalletProvider.ts

```diff
@@ -27,18 +34,57 @@ export class PrivyWalletProvider {
    *   appSecret: "your-app-secret",
    *   chainType: "solana"
    * });
+   *
+   * // For Embedded Wallet
+   * const embeddedWallet = await PrivyWalletProvider.configureWithWallet({
+   *   appId: "your-app-id",
+   *   appSecret: "your-app-secret",
+   *   walletId: "delegated-wallet-id",
+   *   walletType: "embedded"
+   * });
    * ```
    */
   static async configureWithWallet<T extends PrivyWalletConfig>(
-    config: T & { chainType?: "ethereum" | "solana" },
-  ): Promise<T extends { chainType: "solana" } ? PrivySvmWalletProvider : PrivyEvmWalletProvider> {
+    config: T & {
+      chainType?: "ethereum" | "solana";
+      walletType?: "server" | "embedded";
+    },
+  ): Promise<
+    T extends { walletType: "embedded" }
```

Comment:
This type is getting long, and the fact that we have to `as` it for each case below is not ideal.

Can you extract this to a type at the top of the file under `PrivyWalletConfig`? Something like:
```ts
type PrivyWalletProviderImplementation<T> = 
  T extends { walletType: "embedded" } 
    ? PrivyEvmEmbeddedWalletProvider
    : T extends { chainType: "solana" }
      ? PrivySvmWalletProvider
      : PrivyEvmWalletProvider;
```

and then use `PrivyWalletProviderImplementation<T>` as the return type and the `as PrivyWalletProviderImplementation<T>` for returns?